server/api/category/categoryController.js

```
@@ -0,0 +1,67 @@
+const Category = require('./categoryModel');
+const _ = require('lodash');
+const logger = require('../../util/logger');
+
+exports.param = (req, res, next, id) => {
+  Category.findById(id)
+    .exec()
+    .then((category) => {
+      if (!category) {
+        next(New Error('No category with that id'));
```

Did you try it? put an `id` that doesn't exist on you db, and try to get this error message.
Because I think you need to return the error here and not call `next`.

---

It is based on Scott's lesson, I didn't try it on this project but it worked perfectly in the tutorial I made.
You think I should change it?

---

Only if it's not return a nice error to the api client.